### PR TITLE
Acall

### LINKED ISSUE #335: Fix incorrect argument results for ACALL with multiple arguments
This ACALL support was added in issue #226.  This new issue has been reported by Melvyn during use in zvsam development.  I'll add example after more research and development of cut down test case.

### LINKED ISSUE #343: assembler test program zivp.asm with 294,789 lines causes assembler to ignore first 200,005 lines.
assembler test program zivp.asm with 294,789 lines causes assembler to ignore first 200,005 lines.

### PR DESCRIPTION
**Pull request by dsh33782**
mz390 has been updated to fix error when no parms are passed to acall.  The build.bat now runs on local PC.

### CODE REVIEW

**Comment by adelosa** [comment]
Fixes #343
Fixes #335

**Review by abekornelis — APPROVED** [review]
looks good enough to me

File: src/mz390.java [hunk 1]
```diff
@@ -429,10 +429,14 @@
...
 	 ^                           2. Access parms via &NAME(sub1,sub2)                 Example N'&SUB1 = 2, &SUB1(2) = B
 	 *                           3. Use tot_macros to make ACALL and AENTRY hash key unique within a macro
-	 *                           4. Add tz390 macro APARM generated by ACALL to reset parm just before AENTRY
+	 *                           4. Add tz390 opcode APARM generated by ACALL to reset parm just before AENTRY
 	 * 2020-12-28 dsh rpi 2226 change sysvsam default to 1 see tz390 zvsam option
 	 * 2021-01-07 dsh rpi 2228 supress error searching for symbol with option NOASM for ZPARTRS
 	 * 2021-08-11 dsh issue #318 dup of 261 fix identified by John Ganci, let az390 convert sdts to numeric as macro may test for sdt strings
+	 * 2022-01-17 dsh issue #343 abort if maxline exceeded
+	 * 2022-01-25 dsh issue #335 set &(acall)(n) parms at aparm call using zam insert MLC line just before aentry for acall
+	 *            dsh #335 make find_acall_name separate from fine_var using ACALL_ prefix
```
Comment (jyganci) [inline src/mz390.java:438]: fine_var should be find_var.

**Comment by dsh33782** [comment]
John

 

Correct, I’ll fix the comment in v1.8.1 update for next change to mz390.java.  Fortunately, the source is not in rc1.

 

Don Higgins

 ***@***.***> ***@***.***

www.don-higgins.net

File: src/mz390.java [hunk 2]
```diff
@@ -3210,7 +3220,44 @@ private boolean zsm_find_name(){
...
+			   if (!acall_arg.substring(0,1).equals("\'")){  // #335
+		          acall_arg_insert = "&(" + acall_name + ")(" + acall_arg_num + ") SETC \'" + acall_arg.substring(0,acall_arg_comma) + "\'";
+		       } else {
+				  acall_arg_insert = "&(" + acall_name + ")(" + acall_arg_num + ") SETC " + acall_arg.substring(0,acall_arg_comma);
+			   }
+			   acall_arg_num++;
+        	   insert_acall_set(acall_arg_insert);
+               acall_arg = acall_arg.substring(acall_arg_comma+1);
+ 			   acall_arg_comma = acall_arg.indexOf(',');
+		 }
```
Comment (jyganci) [inline src/mz390.java:3243]: Has the string been checked carefully before calling insert_acall_parms? If not, see next items.
1. Line 3234 checks for single quote. If found, it arbitrarily sets ending single quote on line 3235. What if the comma is within a quoted substring?
2. Line 3241 extracts substring following the found comma. What if the string ends with a comma? What if there were two commas in a row?

These may be okay. Just asking.

**Review by jyganci — COMMENTED** [review]
Made a couple of comments. Please review them.

**Pull request merged**